src/coreclr/crossgen-corelib.proj

```diff
@@ -8,18 +8,18 @@
       <UseCrossgen2 Condition="'$(UseCrossgen2)' == ''">false</UseCrossgen2>
 
       <OSPlatformConfig>$(TargetOS).$(TargetArchitecture).$(Configuration)</OSPlatformConfig>
-      <RootBinDir>$(RepoRoot)\artifacts</RootBinDir>
-      <LogsDir>$(RootBinDir)\log</LogsDir>
-      <BinDir>$(RootBinDir)\bin\coreclr\$(OSPlatformConfig)</BinDir>
-      <IntermediatesDir>$(RootBinDir)\obj\coreclr\$(OSPlatformConfig)</IntermediatesDir>
-      <CrossGenCoreLibLog>$(LogsDir)\CrossgenCoreLib_$(TargetOS)__$(TargetArchitecture)__$(Configuration).log</CrossGenCoreLibLog>
+      <RootBinDir>$([MSBuild]::NormalizeDirectory('$(RepoRoot)', 'artifacts'))</RootBinDir>
+      <LogsDir>$([MSBuild]::NormalizeDirectory('$(RootBinDir)', 'log'))</LogsDir>
+      <BinDir>$([MSBuild]::NormalizeDirectory('$(RootBinDir)', 'bin', 'coreclr', $(OSPlatformConfig)))</BinDir>
+      <IntermediatesDir>$([MSBuild]::NormalizeDirectory('$(RootBinDir)', 'obj', 'coreclr', $(OSPlatformConfig)))</IntermediatesDir>
+      <CrossGenCoreLibLog>$([MSBuild]::NormalizePath('$(LogsDir)', 'CrossgenCoreLib_$(TargetOS)__$(TargetArchitecture)__$(Configuration).log'))</CrossGenCoreLibLog>
       <ExeExtension Condition="'$(OS)' == 'Windows_NT'">.exe</ExeExtension>
```

Comment:
It looks useful to have a `ScriptExtension` set to `cmd/sh` similar to `ExeExtension` and replace blocks like
```
<DotNetCli>$([MSBuild]::NormalizePath('$(RepoRoot)', 'dotnet.sh'))</DotNetCli>
<DotNetCli Condition="'$(OS)' == 'Windows_NT'">$([MSBuild]::NormalizePath('$(RepoRoot)', 'dotnet.cmd'))</DotNetCli>
```
with 
```
<DotNetCli>$([MSBuild]::NormalizePath('$(RepoRoot)', 'dotnet.$(ExeExtension)'))</DotNetCli>
```

is there any issue with this?

Reply:
I like this idea, I did a similar thing in the YAML scripts about a year ago ;-).